arrows/ocv/tests/test_image.cxx

```diff
@@ -345,43 +345,77 @@ TYPED_TEST(image_conversion, vital_to_ocv_interleaved)
+};
+
+using bgr_types = ::testing::Types<uint8_t, uint16_t, float>;
+TYPED_TEST_CASE(bgr_conversion, bgr_types);
+
+// ----------------------------------------------------------------------------
+TYPED_TEST(bgr_conversion, bgr_to_rgb)
+{
+  kwiver::vital::image_of<TypeParam> img{ 200, 300, 3 };
+  populate_vital_image<TypeParam>( img );
   cv::Mat ocv_img =  ocv::image_container::vital_to_ocv(img, ocv::image_container::BGR_COLOR);
   {
     int const num_c = ocv_img.channels();
     for( int j = 0; j < ocv_img.rows; ++j )
     {
       for( int i = 0; i < ocv_img.cols; ++i )
       {
-        ASSERT_EQ( img( i, j, 0 ), ocv_img.ptr<unsigned char>( j )[ num_c * i + 2 ] )
-            << "Pixels differ at " << i << ", " << j << ", (0,2)";
-      }
-    }
-    for( int j = 0; j < ocv_img.rows; ++j )
-    {
-      for( int i = 0; i < ocv_img.cols; ++i )
-      {
-        ASSERT_EQ( img( i, j, 1 ), ocv_img.ptr<unsigned char>( j )[ num_c * i + 1 ] )
-            << "Pixels differ at " << i << ", " << j << ", (1,1)";
+        for( int c = 0; c < 3; ++c )
+        {
+          ASSERT_EQ( img( i, j, c ), ocv_img.ptr<TypeParam>( j )[ num_c * i + (2-c) ] )
+              << "Pixels differ at " << i << ", " << j << ", (" << c << "," << (2-c) << ")";
+        }
       }
     }
+  }
+}
+
+// ----------------------------------------------------------------------------
+TYPED_TEST(bgr_conversion, bgra_to_rgba)
+{
+  kwiver::vital::image_of<TypeParam> img{ 200, 300, 4 };
+  populate_vital_image<TypeParam>( img );
+  cv::Mat ocv_img =  ocv::image_container::vital_to_ocv(img, ocv::image_container::BGR_COLOR);
+  {
+    int const num_c = ocv_img.channels();
     for( int j = 0; j < ocv_img.rows; ++j )
     {
       for( int i = 0; i < ocv_img.cols; ++i )
       {
-        ASSERT_EQ( img( i, j, 2 ), ocv_img.ptr<unsigned char>( j )[ num_c * i + 0 ] )
-            << "Pixels differ at " << i << ", " << j << ", (2,0)";
+        for( int c = 0; c < 3; ++c )
+        {
+          ASSERT_EQ( img( i, j, c ), ocv_img.ptr<TypeParam>( j )[ num_c * i + (2-c) ] )
+              << "Pixels differ at " << i << ", " << j << ", (" << c << "," << (2-c) << ")";
+        }
+        ASSERT_EQ( img( i, j, 3 ), ocv_img.ptr<TypeParam>( j )[ num_c * i + (3) ] )
+              << "Pixels differ at " << i << ", " << j << ", (3,3)";
       }
     }
-  };
+  }
 }
 
+// ----------------------------------------------------------------------------
+template <typename T>
+class bgr_bad_conversion : public ::testing::Test
+{
+};
+
+using bgr_bad_types = ::testing::Types<int8_t, int16_t, int32_t, double>;
+TYPED_TEST_CASE(bgr_bad_conversion, bgr_bad_types);
+
+// ----------------------------------------------------------------------------
+TYPED_TEST(bgr_bad_conversion, bgr_to_rgb_bad_types)
+{
+  kwiver::vital::image_of<TypeParam> img{ 200, 300, 3 };
+  populate_vital_image<TypeParam>( img );
+  EXPECT_THROW(ocv::image_container::vital_to_ocv(img, ocv::image_container::BGR_COLOR),
```

Comment:
I'm glad to see this tested, but this might be a little excessive for parameterization :wink:. Would it make sense to unroll and fold this into the existing `TEST(image, bad_conversions)`?

Reply:
It ends up being fewer lines of code.  I am going to leave it this way.

Reply:
Really? I came up with 19 (current) vs. 16 (suggested):

```c++
  EXPECT_THROW( ocv::image_container::vital_to_ocv(
                  image_of<int8_t>{ 200, 300 },
                  ocv::image_container::BGR_COLOR ),
                image_type_mismatch_exception );
// ...likewise for other three types...
```

Reply:
Current is 16.